Marsiple/Common/Network/PostsFetcher.swift

```diff
@@ -8,24 +8,44 @@
 
 import Foundation
 
+enum PostsFetcherError: LocalizedError {
+    case urlCreationFailure
+    case dataUnwrapingFailure
+    case parsingDataFailure
+    case generic(Error)
+    
+    var errorDescription: String? {
+        switch self {
+        case .urlCreationFailure:
+            return "Error creating URL."
+        case .dataUnwrapingFailure:
+            return "Error unwrapping data."
+        case .parsingDataFailure:
+            return "Error parsing data."
+        case .generic(let error):
+            return error.localizedDescription
+        }
+    }
+}
+
 struct PostsFetcher {
-    func get(succes: @escaping ([Post]) -> Void, faliure: @escaping (String) -> Void) {
-        guard let url = URL(string: MartianApi.url + "/posts")
-            else { return DispatchQueue.main.async { faliure("Error creating URL.") } }
+    func get(success: @escaping ([Post]) -> Void, failure: @escaping (LocalizedError) -> Void) {
+        guard let url = MartianApi.url?.appendingPathComponent("/posts")
```

Comment:
Hmm is `/` (within `/posts`) necessary here? I would've thought just `posts` should be enough.

I haven't tried it out, just asking.

Reply:
It could be within `MartianApi.url` but it's related to posts.

Reply:
I mean, the `appendingPathComponent` method should be able to resolve just `posts` as the path, instead of `posts`.

Reply:
Didn't know it adds `/` itself. It says in the method description, if I know this is a directory I should use `appendingPathComponent(pathComponent: #String#, isDirectory:#Bool#)` form.

Reply:
I'm not 💯 sure it adds the `/`, could you please confirm?

Reply:
It does in this case, but I don't know how it recognises it should do that. The description says it preforms some file system operation to determine if it is or if it's not a directory.

Reply:
My guess is that, since this is not a `file system url`, but a `remote url`, the system knows that a `path` needs `/` to work properly.

Reply:
Should I use this form `appendingPathComponent(pathComponent: #String#, isDirectory:#Bool#)`?

Reply:
No, I don't think we should use that, as we're not referring to the `file system`.